### ISSUE TITLE
block-scoped-var doesn't work with const and let

### ISSUE THREAD

**Issue by ffxsam** [eslint/eslint#9311]
ESLint v1.0.2. I noticed `block-scoped-var` only seems to work on `var`, not `const` and `let`. Is this intentional? I enforce `no-var` in my projects, rendering `block-scoped-var` useless—but it's such an easy mistake to make, I want that rule to apply to `let` and `const`.

Sample code:
```js
function thisThing() {
  try {
    const something = 123;
    return something;
  } catch (e) {
    throw new Error(e, something);
  }
}

thisThing();
```

**Comment by eslintbot** [comment]
Hi @ffxsam, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by joovel** [comment]
@ffxsam : `let` and `const` are block-scoped, not function-scoped like `var`. In the case above, the code in the `catch` block should fail the `no-undef` rule. I put your code into the [ESLint Demo](https://eslint.org/demo/) with ECMA version 6 and I get a failure in `no-undef`.

**Comment by ffxsam** [comment]
Ah! Should've been obvious 😄 Thanks!
